src/beacons/use-beacons.ts

```diff
@@ -0,0 +1,209 @@
+import { storage } from '@atb/storage';
+import { useIsBeaconsEnabled } from './use-is-beacons-enabled';
+import { KETTLE_API_KEY } from '@env';
+import { useCallback, useEffect, useState } from 'react';
+import { NativeModules, Platform } from 'react-native';
+import { Kettle, KettleModules } from 'react-native-kettle-module';
+import { KettleConsents } from 'react-native-kettle-module';
+import { PERMISSIONS, Permission, RESULTS, check, checkMultiple, request } from 'react-native-permissions';
+
+enum storeKey {
+  beaconsConsent = '@ATB_beacons_consent_granted',
+}
+
+const BEACONS_CONSENTS = [KettleConsents.SURVEYS, KettleConsents.ANALYTICS];
+
+type PermissionKey = 'bluetooth' | 'location' | 'motion';
+
+const PERMISSIONS_MAP: Record<PermissionKey, Permission> = {
+  bluetooth: Platform.OS == 'ios' ? PERMISSIONS.IOS.BLUETOOTH_PERIPHERAL : PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION,
+  location: Platform.OS == 'ios' ? PERMISSIONS.IOS.LOCATION_ALWAYS : PERMISSIONS.ANDROID.ACCESS_BACKGROUND_LOCATION,
+  motion: Platform.OS == 'ios' ? PERMISSIONS.IOS.MOTION : PERMISSIONS.ANDROID.ACTIVITY_RECOGNITION,
+};
+
+export type KettleInfo = {
+  isKettleStarted: boolean;
+  kettleIdentifier: string | null;
+  kettleConsents: Record<string, boolean> | null;
+  isBeaconsOnboarded: boolean;
+  privacyDashboardUrl?: string;
+  privacyTermsUrl?: string;
+};
+
+export const useBeacons = () => {
+  const [isBeaconsEnabled, debugOverrideReady] = useIsBeaconsEnabled();
+  const [kettleInfo, setKettleInfo] = useState<KettleInfo>();
+  // The kettle SDK was initialized
+  const [isKettleSDKInitialized, setKettleSDKInitialized] = useState(false);
+  // The app was compiled to support beacons and KEYTLE_API_KEY is set
+  const isBeaconsSupported = (isBeaconsEnabled && debugOverrideReady) && !!KETTLE_API_KEY;
+
+  const updateKettleInfo = useCallback(async () => {
+    if (isBeaconsSupported && isKettleSDKInitialized) {
+      const status = await Kettle.isStarted();
+      const identifier = await Kettle.getIdentifier();
+      const consents = await Kettle.getGrantedConsents();
+      const privacyDashboardUrl = await Kettle.getPrivacyDashboardUrl();
+      const privacyTermsUrl = await Kettle.getPrivacyTermsUrl();
+
+      setKettleInfo({
+        isKettleStarted: status,
+        kettleIdentifier: identifier,
+        kettleConsents: consents,
+        isBeaconsOnboarded: Object.keys(consents).length > 0,
+        privacyDashboardUrl,
+        privacyTermsUrl,
+      });
+    }
+  }, [setKettleInfo, isBeaconsSupported, isKettleSDKInitialized]);
+
+  // This function must be called before consent is granted
+  const initializeBeaconsSDK = useCallback(async () => {
+    // This will set kettle if KETTLE_API_KEY is set on the native side
+    if (!isKettleSDKInitialized && isBeaconsSupported) {
+      await NativeModules.KettleSDKExtension.initializeKettleSDK();
```

Comment:
I think this should also run
`const permissions = await allowedPermissionForKettle();`
and then init only if `permissions.length > 0`

Reply:
It is a double check, because is supposed that if the user went that far is because at least Bluetooth was accepted.